### PR TITLE
Update charon

### PR DESCRIPTION
**Pull request by Nadrieril**
Companion PR to https://github.com/AeneasVerif/charon/pull/308

### CODE REVIEW

File: compiler/SymbolicToPure.ml [hunk 1]
```diff
@@ -578,7 +578,7 @@ let translate_type_decl_kind (span : Meta.span) (kind : T.type_decl_kind) :
   | Alias _ ->
       craise __FILE__ __LINE__ span
         "type aliases should have been removed earlier"
-  | T.Opaque -> Opaque
+  | T.Opaque | T.Error _ -> Opaque
```
Comment (sonmarcho) [inline compiler/SymbolicToPure.ml:581]: We should raise an exception in the Error case

Comment (Nadrieril) [reply compiler/SymbolicToPure.ml:581]: The exception has already been raised by charon. If we're still here we may as well process as much as possible.

Comment (Nadrieril) [reply compiler/SymbolicToPure.ml:581]: Btw this can only happen with `charon --errors-as-warnings`

Comment (sonmarcho) [reply compiler/SymbolicToPure.ml:581]: Ok, makes sense

**Pull request merged**